src/core/session/session-manager.ts

```diff
@@ -293,6 +293,14 @@ export class SessionManager {
         return undefined;
     }
 
+    /**
+     * Returns an in-memory session without loading from storage.
+     * Useful for fast checks (e.g., cancellation) when we do not want to create or restore a session.
+     */
+    public peekSession(sessionId: string): ChatSession | undefined {
+        return this.sessions.get(sessionId);
+    }
```

Comment:
add an argument `restoreFromStorage?: boolean` into existing `getSession` and handle this there instead of making this new method - allows any use case to get a session and decide if they want to pull from memory only or storage as well